site/index.js

```diff
@@ -0,0 +1,352 @@
+'use strict'
+
+// Utility functions:
+
+function post(path, dataObj) {
+  if (!serverURL.value) {
+    return {error: 'client error - server URL not specified yet'}
```

Comment:
Surely this should be `return Promise.reject({...})`?

Reply:
Probably. It should be noted that server errors also return data ala `{error: 'asdf'}`. Maybe we could make `fetchHelper` automatically throw responses which include an `error` field?

Reply:
Fixed. Errors are now a lot more error-y. This also simplified the code a fair amount!